pandas/tests/indexes/test_interval.py

```diff
@@ -395,25 +597,129 @@ def slice_locs_decreasing_cases(self, tuples):
         assert slice_locs[0] == slice_locs[1]
 
     def test_slice_locs_decreasing_int64(self):
-        self.slice_locs_cases([(2, 4), (1, 3), (0, 2)])
+        ### THIS METHOD TO BE REMOVED FOR BEHAVIOR UPDATE IN 0.21
+        self.slice_locs_decreasing_cases([(2, 4), (1, 3), (0, 2)])
 
     def test_slice_locs_decreasing_float64(self):
-        self.slice_locs_cases([(2., 4.), (1., 3.), (0., 2.)])
+        ### THIS METHOD TO BE REMOVED FOR BEHAVIOR UPDATE IN 0.21
+        self.slice_locs_decreasing_cases([(2., 4.), (1., 3.), (0., 2.)])
 
     def test_slice_locs_fails(self):
+        ### THIS METHOD TO BE REMOVED FOR BEHAVIOR UPDATE IN 0.21
         index = IntervalIndex.from_tuples([(1, 2), (0, 1), (2, 3)])
         with pytest.raises(KeyError):
             index.slice_locs(1, 2)
 
-    def test_get_loc_interval(self):
-        assert self.index.get_loc(Interval(0, 1)) == 0
-        assert self.index.get_loc(Interval(0, 0.5)) == 0
-        assert self.index.get_loc(Interval(0, 1, 'left')) == 0
-        pytest.raises(KeyError, self.index.get_loc, Interval(2, 3))
-        pytest.raises(KeyError, self.index.get_loc,
-                      Interval(-1, 0, 'left'))
+    ### NEW SLICE LOCS BEHAVIOR
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def slice_locs_with_interval_updated_behavior(self):
+        # @jreback might prefer these be broken up into separate methods
+
+        # increasing
+        index = IntervalIndex.from_tuples([(0, 2), (1, 3), (2, 4)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == (0, 1)
+        assert index.slice_locs(start=Interval(0, 2)) == (0, 1, 2)
+        assert index.slice_locs(end=Interval(2, 4)) == (0, 1)
+        assert index.slice_locs(end=Interval(0, 2)) == ()
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == ()
+
+        # decreasing
+        index = IntervalIndex.from_tuples([(2, 4), (1, 3), (0, 2)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == ()
+        assert index.slice_locs(start=Interval(0, 2)) == (2)
+        assert index.slice_locs(end=Interval(2, 4)) == ()
+        assert index.slice_locs(end=Interval(0, 2)) == (0, 1)
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == (0, 1)
+
+        # sorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (0, 2), (2, 4)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == # ?
+        assert index.slice_locs(start=Interval(0, 2)) == # ?
+        assert index.slice_locs(end=Interval(2, 4)) == # ?
+        assert index.slice_locs(end=Interval(0, 2)) == # ?
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == # ?
+
+        # unsorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (2, 4), (0, 2)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == # ?
```

Comment:
Again, I am just comparing with integers, I think this should raise when one of start/end is the duplicate interval, but it can work if you pass a unique key:

```
In [26]: pd.Index([1,2,1]).slice_locs(1)
...
KeyError: 'Cannot get left slice bound for non-unique label: 1'

In [27]: pd.Index([1,2,1]).slice_locs(2)
Out[27]: (1, 3)
```